proxy_presentation.php

```diff
@@ -33,13 +33,17 @@
     exit();
 }
 
+define('RECORD_ID_REGEX', '/[a-f0-9]{40}-[0-9]{13}/');
+preg_match(RECORD_ID_REGEX, $relativepath, $matches);
 
 // If path is of format /presentation/{recording-id}/xxxx we need to check
 // user has access to the recording in question. Urls of this format contain
 // sensitive media.
-$pathparts = explode('/', $relativepath);
-if (isset($pathparts[1]) && $pathparts[1] === 'presentation' && isset($pathparts[2])) {
-    $recordingid = $pathparts[2];
+if (!empty($matches)) {
```

Comment:
I think we should still leave a little more protection here against weird dodgy page proxies, so perhaps we should keep the 'presentation/recordid' check in another little strpos.

Reply:
otherwise this may trip on a URL that isnt necessarily a presentation URL. Super edge case, just a thought